src/main/java/hudson/plugins/emailext/EmailRecipientUtils.java

```diff
@@ -38,28 +38,26 @@
         if (!StringUtils.isBlank(recipientList)) {
             final String expandedRecipientList = fixupDelimiters(envVars.expand(recipientList));
             InternetAddress[] all = InternetAddress.parse(expandedRecipientList.replace("bcc:", "").replace("cc:", ""));
+            // Based upon "Parse the given comma separated sequence of addresses..." from:
+            //   https://javaee.github.io/javamail/docs/api/javax/mail/internet/InternetAddress.html#parse-java.lang.String-
+            String[] recipients = StringUtils.stripAll(expandedRecipientList.split(","));
+            if (recipients.length != all.length) {
+                throw new IllegalArgumentException("Unsupported format in expanded recipient list string: " + expandedRecipientList);
+            }
             final Set<InternetAddress> to = new LinkedHashSet<>();
             final Set<InternetAddress> cc = new LinkedHashSet<>();
             final Set<InternetAddress> bcc = new LinkedHashSet<>();
             final String defaultSuffix = ExtendedEmailPublisher.descriptor().getDefaultSuffix();
 
-            for(InternetAddress address : all) {
```

Comment:
I think what this code was trying to support (badly) is the weird format some stuff returns email address in ie
`"Mouse, Mickey" <mickey@disney.com>`
I just added a test case for this and it breaks with your changes

Reply:
But thanks for taking the time to try and fix

Reply:
Oh! That explains the seemingly overly complex/confusing existing (and with respect to JENKINS-52748 buggy) code: thanks for the hint and the new test case. This does not make it really easier ;-)
I might have another look and try at this tomorrow; unless you already have an idea that you want to implement...